sdk/identity/Azure.Identity/src/Credentials/EnvironmentCredential.cs

```diff
@@ -85,6 +92,7 @@ internal EnvironmentCredential(CredentialPipeline pipeline, TokenCredentialOptio
                         AuthorityHost = _options.AuthorityHost,
                         IsLoggingPIIEnabled = _options.IsLoggingPIIEnabled,
                         Transport = _options.Transport,
+                        AdditionallyAllowedTenantsCore = new List<string>(_options.AdditionallyAllowedTenantsCore),
```

Comment:
The `EnvironmentCredential` doesn't do any authentication when `GetToken(Async)` is called. It farms that call out to the underlying credential which are what ultimately authenticates, and perform the tenant id validations. In the case the underlying credential is `ClientSecretCredential` or `UsernamePassword` credential we can pass `_options` directly through, but in the case of `ClientCertificateCredential` we need to possibly send an option which is specific to `ClientCertificateCredentialOptions`, so since we can't clone the options class to the more specific subclass, we're manually copying the relevent options to a new instance of `ClientCertificateCredentialOptions`. We could add some cloning mechanisms which would allow us to create a more specific subclass of options copying all the properties, but this would require some new APIs in Azure.Core.